packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3920,7 +3920,11 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     }
     properties.add(FlagProperty('dirty', value: dirty, ifTrue: 'dirty'));
     if (_dependencies != null && _dependencies.isNotEmpty) {
-      final List<DiagnosticsNode> diagnosticsDependencies = _dependencies
+      final List<InheritedElement> sortedDependencies = _dependencies
+        .toList()
+        ..sort((InheritedElement a, InheritedElement b) =>
+          a.widget.runtimeType.toString().compareTo(b.widget.runtimeType.toString()));
```

Comment:
There is a helper method that @dnfield wrote (I think?) to avoid accessing widget.runtimeType.toString directly, which is slow.

Reply:
it's in foundation - but I don't think it would help here, because you're actually trying to sort the string results here...

In theory this is only called from debug code.

I strongly suspect that if the tests care about order, this isn't helping anything. And if they don't care about order, we should probably just make them tolerate out of order items rather than putting this in the framework though.

Unless we just stop using a `HashSet` where we say we're using a `Set` :)

Reply:
This is the test that cares about order:
https://github.com/flutter/flutter/blob/a1fa1a3d261f5cee1e8dd2e7a67132f465054d83/packages/flutter/test/material/stepper_test.dart#L536
The stringified `Stepper` widget displays its dependencies.
According to the test I added, it should help.